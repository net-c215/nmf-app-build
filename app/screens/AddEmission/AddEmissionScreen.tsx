import React, { useState } from "react";
import { View, ScrollView } from "react-native";

import { Text, Tag } from "../../components";
import styles from "./AddEmissionScreen.styles";
import navigationOptions from "./AddEmissionScreen.navigationOptions";
import { t } from "../../utils/translations";
import { Food, Transport, Custom, AddEmissionButton } from "./components";
import { EmissionTypeEnum } from "../../interfaces";

const AddEmissionScreen = ({ navigation }) => {
  const [emissionType, SetEmissionType] = useState(EmissionTypeEnum.transport);

  return (
    <ScrollView style={styles.container}>
      <View style={styles.typeContainer}>
        <View style={styles.textContainer}>
          <Text.H3>{t("ADD_EMISSION_EMISSION_TYPE")}</Text.H3>
        </View>
        <ScrollView horizontal style={styles.tagContainer}>
          <Tag
            icon={"md-airplane"}
            selected={emissionType === EmissionTypeEnum.transport}
            title={"Transport"}
            onPress={() => SetEmissionType(EmissionTypeEnum.transport)}
          />
          <Tag
            icon={"md-restaurant"}
            selected={emissionType === EmissionTypeEnum.food}
            title={"Food"}
            onPress={() => SetEmissionType(EmissionTypeEnum.food)}
          />
          <Tag
            icon={"md-build"}
            selected={emissionType === EmissionTypeEnum.custom}
            title={"Custon"}
            onPress={() => SetEmissionType(EmissionTypeEnum.custom)}
          />
        </ScrollView>
      </View>

      {emissionType === EmissionTypeEnum.transport ? <Transport /> : null}
      {emissionType === EmissionTypeEnum.food ? <Food /> : null}
      {emissionType === EmissionTypeEnum.custom ? <Custom /> : null}

      <AddEmissionButton navigation={navigation} emissionType={emissionType} />
    </ScrollView>
  );
};

AddEmissionScreen.navigationOptions = navigationOptions;

export default AddEmissionScreen;
