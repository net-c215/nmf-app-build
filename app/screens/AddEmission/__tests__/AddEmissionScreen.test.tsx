import React from "react";
import renderer from "react-test-renderer";
import AddEmissionScreen from "../AddEmissionScreen";

const props = {
  navigation: {
    goBack: jest.fn()
  }
};

it("MonthlyBudgetScreen renders correctly", () => {
  const tree = renderer.create(<AddEmissionScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("MonthlyBudgetScreen should go back if save button is pressed", () => {
  const root = renderer.create(<AddEmissionScreen {...props} />).root;
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  // @ts-ignore
  const button = root.findByType("Button.Primary");

  button.props.onPress();
  expect(props.navigation.goBack).toHaveBeenCalled();
});
