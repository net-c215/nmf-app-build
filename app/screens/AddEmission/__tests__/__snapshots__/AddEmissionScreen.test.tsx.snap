// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEmissionScreen renders correctly 1`] = `
<FormattedProvider
  locale="en"
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "alignContent": "flex-start",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "height": 16,
          }
        }
      />
      <TextInput
        isOptional={true}
        onChangeText={[Function]}
        placeholder="e.g burger, flight from Paris to Romeâ€¦"
        title="Name this emission "
        value=""
      />
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
        />
      </Modal>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          }
        }
      >
        <Text.H3>
          Date
        </Text.H3>
        <View
          style={
            Object {
              "flexDirection": "column",
              "paddingTop": 4,
            }
          }
        >
          <TouchableOpacity
            onPress={[Function]}
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text.Primary
              lightGray={true}
            >
              Sunday 7th April 2019
            </Text.Primary>
            <Text.Primary
              lightGray={true}
            >
               - 
            </Text.Primary>
            <Text.Primary
              bold={true}
              green={true}
            >
              Change
            </Text.Primary>
          </TouchableOpacity>
        </View>
      </View>
      <Button.Primary
        onPress={[Function]}
        style={
          Object {
            "margin": 16,
          }
        }
        textType="Primary"
      >
        <Text.Primary
          bold={true}
          center={true}
          white={true}
        >
          Add this emission
        </Text.Primary>
      </Button.Primary>
    </View>
  </RCTScrollView>
</FormattedProvider>
`;
